server/channels/app/platform/service.go

```diff
@@ -381,6 +381,7 @@ func (ps *PlatformService) ShutdownConfig() error {
 
 func (ps *PlatformService) SetTelemetryId(id string) {
 	ps.telemetryId = id
+	ps.regenerateClientConfig()
```

Comment:
This feels like making the method do more than it's supposed to do. Can we rename the method to reflect what it's doing?

Reply:
Now you pointed out a very hard problem :) I'm open to suggestions.

Reply:
These are doing two very different things really. So hard to give it a single name. Maybe as an alternative, we can have a `PostTelemetryIdHook` function that we can define in platform service, and call `regenerateClientConfig` from there? That way the code becomes self-explaining.